File: changelog_builder/builder.py
class Builder:


    def __init__(self, log_parser):
        self.log_parser = log_parser
        self.previous_log = {}
        self.current_log = {}

    def set_previous_log(self, changelog):
        self.previous_log = self.log_parser.parse(changelog)

    def set_current_log(self, changelog):
        self.current_log = self.log_parser.parse(changelog)

    def get_diff(self):
        diff_keys = set(self.current_log) - set(self.previous_log.keys())

        diff = {}
        for key in diff_keys:
            diff[key] = self.current_log[key]

        return diff

File: lib/builder.js
var config = require('../config'),
	LogParser = require('./parser/' + config.vcs.type + 'log')
	ArrayHelper = require('./util/arrayhelper');

var logParser = new LogParser(),
	arrayHelper = new ArrayHelper();

function Builder() {
	this.oldLog = {};
	this.newLog = {};
}

Builder.prototype.setOldLog = function(log) {
	this.oldLog = logParser.parse(log);

	return true;
}

Builder.prototype.setNewLog = function(log) {
	this.newLog = logParser.parse(log);

	return true;
}

Builder.prototype.getDiff = function() {
	var diffKeys = arrayHelper.diff(Object.keys(this.newLog), Object.keys(this.oldLog));

	var diff = {};
	for (var i in diffKeys) {
		var key = diffKeys[i];
		diff[key] = this.newLog[key]
	}

	return diff;
}

module.exports = Builder;
